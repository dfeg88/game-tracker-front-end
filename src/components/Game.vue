<template>
  <v-card :class="{'ma-3': $vuetify.breakpoint.smAndDown, 'ma-3': $vuetify.breakpoint.mdAndUp}"
          class="elevation-8 mb-2">
    <v-card-title id="game-name" class="justify-center">{{ game.name }}</v-card-title>
    <v-img
        :src="getBoxArt(game.boxArtPath)"
        height="200px"
        id="boxArt"
        :contain="true">
      <v-container
          fill-height
          fluid
          pa-2>
      </v-container>
    </v-img>
    <v-card-text>
      <p id="publisher"><b>Publisher:</b> {{ game.publisher }}</p>
      <p id="platform"><b>Platform:</b> {{ game.platform }}</p>
      <p id="release-date"><b>Release Date:</b> {{ new Date(game.releaseDate).toDateString() }}</p>
      <p id="number-of-players"><b>Number of Players:</b> {{ game.numberOfPlayers }}</p>
      <p id="genre"><b>Genre:</b> {{ game.genre }}</p>
    </v-card-text>
  </v-card>
</template>

<script>
export default {
  name: "Game",
  props: {
    game: Object
  },
  methods: {
    getBoxArt(boxArtPath) {
      return `http://localhost:4566/game-tracker/${boxArtPath}`
    }
  }
}
</script>