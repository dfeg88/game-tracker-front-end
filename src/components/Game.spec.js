import {createLocalVue, shallowMount} from "@vue/test-utils";
import Vuetify from "vuetify";
import Game from "@/components/Game";

let wrapper;
const vuetify = new Vuetify();

describe('Game', function () {

  beforeEach(() => {
    wrapper = shallowMount(Game, {
      localVue: createLocalVue(),
      vuetify,
      propsData: {
        game: {
          name: 'name',
          platform: 'platform',
          genre: 'genre',
          releaseDate: '01-01-2020',
          numberOfPlayers: 8,
          publisher: 'publisher',
        }
      }
    })
  })

  it('renders expected HTML structure', () => {
    expect(wrapper.elements).toMatchSnapshot();
  })

  it('renders values from props', () => {
    expect(wrapper.find('#game-name').text()).toBe('name');
    expect(wrapper.find('#platform').text()).toBe('Platform: platform');
    expect(wrapper.find('#genre').text()).toBe('Genre: genre');
    expect(wrapper.find('#release-date').text()).toBe('Release Date: Wed Jan 01 2020');
    expect(wrapper.find('#number-of-players').text()).toBe('Number of Players: 8');
    expect(wrapper.find('#publisher').text()).toBe('Publisher: publisher');
  })
});