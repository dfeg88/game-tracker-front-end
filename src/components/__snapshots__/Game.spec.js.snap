// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Game renders expected HTML structure 1`] = `undefined`;
